src/mlpack/tests/init_rules_test.cpp

```diff
@@ -126,6 +129,35 @@ BOOST_AUTO_TEST_CASE(OivsInitTest)
   BOOST_REQUIRE_EQUAL(1, 1);
 }
 
+BOOST_AUTO_TEST_CASE(XavierInitTest)
+{
+  arma::mat weights;
+  //RandomSeed(21);
+  XavierInit<XavierUniform> xavierUniform;
+  XavierInit<XavierNormal> xavierNormal;
+  xavierUniform.Initialize(weights, 100, 100);
+  bool b = arma::all(arma::vectorise(weights) >= -(1e-2) || arma::vectorise(weights) <= (1e-2));
+  BOOST_REQUIRE_EQUAL(b, 1);
+  xavierNormal.Initialize(weights, 100, 100);
+  //std::cout << weights << std::endl;
+  auto ret = Shapiro(weights, 0.05);
+  BOOST_REQUIRE_EQUAL(ret.accept, true);
+}
+
+BOOST_AUTO_TEST_CASE(HeInitTest)
+{
+  arma::mat weights;
+  //RandomSeed(21);
+  HeInit<HeUniform> Uniform;
+  HeInit<HeNormal> Normal;
+  Uniform.Initialize(weights, 100, 100);
+  bool b = arma::all(arma::vectorise(weights) >= -(1e-2) || arma::vectorise(weights) <= (1e-2));
+  BOOST_REQUIRE_EQUAL(b, 1);
+  Normal.Initialize(weights, 100, 100);
+  auto ret = Shapiro(weights, 0.05);
+  BOOST_REQUIRE_EQUAL(ret.accept, true);
```

Comment:
Could you split these into two tests please? Better to test each thing separately.